packages/console/src/components/IconButton/index.tsx

```diff
@@ -1,18 +1,49 @@
+import { AdminConsoleKey } from '@logto/phrases';
+import { Nullable } from '@silverhand/essentials';
 import classNames from 'classnames';
-import { HTMLProps } from 'react';
+import { ForwardedRef, forwardRef, HTMLProps, useImperativeHandle, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
 
+import Tooltip from '../Tooltip';
 import * as styles from './index.module.scss';
 
 export type Props = Omit<HTMLProps<HTMLButtonElement>, 'size' | 'type'> & {
   size?: 'small' | 'medium' | 'large';
+  tooltip?: AdminConsoleKey;
 };
 
-const IconButton = ({ size = 'medium', children, className, ...rest }: Props) => {
+const IconButton = (
+  { size = 'medium', children, className, tooltip, ...rest }: Props,
+  reference: ForwardedRef<HTMLButtonElement>
+) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const innerReference = useRef<HTMLButtonElement>(null);
+
+  useImperativeHandle<Nullable<HTMLButtonElement>, Nullable<HTMLButtonElement>>(
```

Comment:
Can we remove this function?

Reply:
We need to use this function with `forwardRef` function so that we can reuse the `forwardedRef`.

https://reactjs.org/docs/hooks-reference.html#useimperativehandle

Reply:
The `reference: ForwardedRef<HTMLButtonElement>` and `innerReference: RefObject<HTMLButtonElement>` are different things.
We use `useImperativeHandle` to reuse the `reference: ForwardedRef<HTMLButtonElement>`.
And by doing this, when we visit the parent's `RefObject.current`, which will forwarded by the `forwardRef` function, we will get `innerRef.current`.